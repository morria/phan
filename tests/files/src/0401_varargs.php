<?php

function test_varargs401(stdClass $a1, stdClass ...$args) {
    var_export($args);
}

/** @param array<string,stdClass> $args */
function unpack_varargs401(array $args, iterable $iterable) {
    $c = [2];
    $int = 2;
    $stdClass = new stdClass();
    $arrayObject = new ArrayObject([new stdClass(), new stdClass()]);
    test_varargs401(...$args);
    test_varargs401(...$c);
    test_varargs401(...$int);
    test_varargs401(...$stdClass);
    test_varargs401(...$iterable);
    test_varargs401(...$arrayObject);
}
