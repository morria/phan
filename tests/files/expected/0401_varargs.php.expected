%s:13 PhanTypeMismatchUnpackKey When unpacking a value of type array<string,\stdClass>, the value's keys were of type string, but the keys should be consecutive integers starting from 0
%s:14 PhanTypeMismatchArgument Argument 1 (a1) is 2 but \test_varargs401() takes \stdClass defined at %s:3
%s:15 PhanTypeMismatchUnpackValue Attempting to unpack a value of type 2 which does not contain any subtypes of iterable (such as array or Traversable)
%s:16 PhanTypeMismatchUnpackValue Attempting to unpack a value of type \stdClass which does not contain any subtypes of iterable (such as array or Traversable)
